PokemonLibrary/Features/PokemonDetail/DetailView.swift

```diff
@@ -0,0 +1,64 @@
+//
+//  DetailView.swift
+//  PokemonLibrary
+//
+//  Created by Jessi Febria on 06/05/25.
+//
+
+import Foundation
+import SwiftUI
+
+struct DetailView: View {
+    @StateObject var viewModel: DetailViewModel
+
+    var body: some View {
+        VStack {
+            switch viewModel.state {
+            case .loading:
+                ProgressView("Catching Pokémon details...")
+                    .progressViewStyle(.circular)
+            case .loaded(detail: let pokemon):
+                AsyncImage(
+                    url: URL(string: pokemon.sprites.other.officialArtwork.url)
+                ) { image in
+                    image
+                        .resizable()
+                        .aspectRatio(contentMode: .fit)
+                        .frame(width: 200, height: 200)
+                } placeholder: {
+                    ProgressView()
+                }
```

Comment:
nice catch to use AsyncImage! bisa nih di-enhance dengan handle untuk error state-nya juga


```suggestion
                 AsyncImage(url: pokemon.imageUrl) { phase in
                        switch phase {
                        case .empty:
                            ProgressView()
                        case .success(let image):
                            image
                                .resizable()
                        default:
                            Image(systemName: "xmark.octagon")
                        }
                    }
                    .frame(width: 200.0, height: 200.0)
                    .scaledToFill()
                    .clipped()
```